lib/VolSurface/Utils.pm

```diff
@@ -140,6 +144,14 @@ sub get_strike_for_spot_delta {
         croak 'Provided delta [' . $new_args{delta} . '] must be on [0,1]';
     }
 
+    if ($new_args{forward}) {
+        if ($new_args{premium_adjusted}){
+            $new_args{delta} = $new_args{delta} * exp(-$new_args{r_rate} * $new_args{t});
+        } else {
+            $new_args{delta} = $new_args{delta} * exp(-$new_args{q_rate} * $new_args{t});
+        }
+    }
+
```

Comment:
As per the previous comments, it turns out that this entire if statement can be (re)condensed into

```perl
if ($new_args{forward}) {
    $new_args{delta} = $new_args{delta} * exp(-$new_args{q_rate} * $new_args{t});
}
```

Basically just the else condition.